### PR TITLE
Navigate to separate verification page for oob_otp

### PR DESCRIPTION
**Pull request by pkong-ds**
## Blocked by
- #4474 
- #4475 

## What's this PR?
Authgear currently auto-picks `index=0` (first) option branch. If the first branch is `oob-otp`, email/otp are directly send on previous step finished. 

This PR allows user input between
- previous step finish
- next step auto-pick

For more context on UX, see [slack thread](https://oursky.slack.com/archives/C03AH1ACM/p1721291221300689)

## Preview 1 - `primary_oob_otp_email`

https://github.com/user-attachments/assets/96d72b27-ef64-4b3a-8d87-1e672aa574ce

## Preview 2 - `primary_oob_otp_sms`

https://github.com/user-attachments/assets/9d1bd898-b4ca-41e0-8aa8-6464638f8d1b

## Preview 3 - `secondary_oob_otp_email`

https://github.com/user-attachments/assets/20b4ac35-76d7-4a60-ab5d-8ead578f9b07

## Preview 4 - `secondary_oob_otp_sms`

https://github.com/user-attachments/assets/bb02f2f2-f393-48a7-b34f-4bb9d19190d9

### CODE REVIEW

**Comment by louischan-oursky** [comment]
1. Rebase `feat-bot-protection-ui` onto `main`
2. Rebase this PR onto `feat-bot-protection-ui`

**Comment by pkong-ds** [comment]
> Rebase feat-bot-protection-ui onto main

No access 😢 can help?

```
Pushing to github.com:authgear/authgear-server.git
ERROR: Permission to authgear/authgear-server.git denied to pkong-ds.
fatal: Could not read from remote repository.
Please make sure you have the correct access rights
and the repository exists.


In a case you entered incorrect password, please
update it in Keychain Access application.
```

File: authui/src/authflowv2.ts [hunk 1]
```diff
@@ -108,6 +109,10 @@ Stimulus.register("click-to-switch", ClickToSwitchController);
 Stimulus.register("inline-preview", InlinePreviewController);
 Stimulus.register("previewable-resource", PreviewableResourceController);
 Stimulus.register("bot-protection-input", BotProtectionInputController);
```
Comment (louischan-oursky) [inline authui/src/authflowv2.ts:111]: "bot-protection-token-input"

File: authui/src/authflowv2.ts [hunk 2]
```diff
@@ -108,6 +109,10 @@ Stimulus.register("click-to-switch", ClickToSwitchController);
...
+Stimulus.register(
+  "bot-protection-on-success-click",
```
Comment (louischan-oursky) [inline authui/src/authflowv2.ts:113]: "bot-protection-standalone-page"

**Review by louischan-oursky — APPROVED** [review]

**Pull request merged**